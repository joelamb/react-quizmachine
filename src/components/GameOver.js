import React from 'react';
import '../styles/gameover.scss'

const GameOver = ({ hiscores, score, playerName, setPlayerName, submitHiScore }) => (
  <section className="gameover">
    <h2>Game Over</h2>
    <h3>Leaderboard</h3>
    <table>
      <thead>
        <tr>
          <td>Name</td>
          <td>Score</td>
        </tr>
      </thead>
      <tbody>
        {hiscores.map(item => {
          const { name, score } = item;
          return <tr key={name}>
            <td>{name}</td>
            <td>{score}</td>
          </tr>
        })}
      </tbody>
    </table>
    <form onSubmit={e => {
      e.preventDefault();
      submitHiScore();
    }}
      className="score">
      <fieldset>
        <legend>Save your score</legend>
        <p>
          <label htmlFor="score">Your score:
        <input type="number" name="score" id="score" value={score} readOnly />
          </label>
        </p>
        <p>
          <label htmlFor="playerName">Enter your initials:
        <input
              type="text"
              name="playerName"
              id="playerName"
              onChange={e => setPlayerName(e.target.value)}
              value={playerName}
              pattern="[A-Za-z]{3}"
              placeholder="3 characters"
              className="name" />
          </label></p>
        <button type="submit" className="btn btn__primary">Save score</button>
      </fieldset>
    </form>
  </section>
)

export default GameOver;
