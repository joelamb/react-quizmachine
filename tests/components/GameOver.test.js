import React from 'react'
import GameOver from '../../src/components/GameOver';
import renderer from 'react-test-renderer';

describe('GameOver', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<GameOver hiscores={[{ name: "joe", score: 24 }, { name: "ben", score: 18 }]} />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})