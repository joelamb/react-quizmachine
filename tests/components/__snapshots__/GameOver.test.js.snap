// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver should render correctly 1`] = `
<section
  className="gameover"
>
  <h2>
    Game Over
  </h2>
  <h3>
    Leaderboard
  </h3>
  <table>
    <thead>
      <tr>
        <td>
          Name
        </td>
        <td>
          Score
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          joe
        </td>
        <td>
          24
        </td>
      </tr>
      <tr>
        <td>
          ben
        </td>
        <td>
          18
        </td>
      </tr>
    </tbody>
  </table>
  <form
    className="score"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Save your score
      </legend>
      <p>
        <label
          htmlFor="score"
        >
          Your score:
          <input
            id="score"
            name="score"
            readOnly={true}
            type="number"
            value={undefined}
          />
        </label>
      </p>
      <p>
        <label
          htmlFor="playerName"
        >
          Enter your initials:
          <input
            className="name"
            id="playerName"
            name="playerName"
            onChange={[Function]}
            pattern="[A-Za-z]{3}"
            placeholder="3 characters"
            type="text"
            value={undefined}
          />
        </label>
      </p>
      <button
        className="btn btn__primary"
        type="submit"
      >
        Save score
      </button>
    </fieldset>
  </form>
</section>
`;
